### PR TITLE
More unit test coverage

### PR DESCRIPTION
**Pull request by jomuel**

### CODE REVIEW

File: raiden/tests/unit/test_blockchain_events_handler.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+from raiden_contracts.constants import ChannelEvent
+
+from raiden.blockchain.events import Event
+from raiden.blockchain_events_handler import create_update_transfer_state_change
+from raiden.tests.utils import factories
+from raiden.transfer.state_change import ContractReceiveUpdateTransfer
+
+
+@pytest.fixture
+def container():
```
Comment (palango) [inline raiden/tests/unit/test_blockchain_events_handler.py]: I'd rather have this called `chain_state`

File: raiden/blockchain_events_handler.py [hunk 2]
```diff
@@ -252,16 +252,14 @@ def handle_channel_withdraw(raiden: "RaidenService", event: Event):
         update_path_finding_service_from_channel_state(raiden=raiden, channel_state=channel_state)
 
 
-def handle_channel_closed(raiden: "RaidenService", event: Event):
+def create_channel_closed_state_change(chain_state: "ChainState", event: Event):
```
Comment (palango) [inline raiden/blockchain_events_handler.py]: Please add a return type annotation

File: raiden/blockchain_events_handler.py [hunk 3]
```diff
@@ -295,7 +292,15 @@ def handle_channel_closed(raiden: "RaidenService", event: Event):
             block_number=block_number,
             block_hash=block_hash,
         )
-        raiden.handle_and_track_state_change(route_closed)
+        return route_closed
+
+
+def handle_channel_closed(raiden: "RaidenService", event: Event):  # pragma: no unittest
```
Comment (palango) [inline raiden/blockchain_events_handler.py]: Not sure if this refactoring is a big win...

File: raiden/tests/unit/test_blockchain_events_handler.py [hunk 4]
```diff
@@ -51,3 +57,33 @@ def test_create_update_transfer_state_change_unknown_channel(container, update_t
         chain_state=container.chain_state, event=update_transfer
     )
     assert state_change is None
+
+
+@pytest.fixture
+def channel_closed(container, event_data):
```
Comment (palango) [inline raiden/tests/unit/test_blockchain_events_handler.py]: I'd rename this: `channel_closed` -> `channel_closed_event`, same for all other event fixtures

File: raiden/tests/unit/transfer/test_channel.py [hunk 5]
```diff
@@ -237,3 +257,141 @@ def test_is_balance_proof_usable_onchain_answer_is_false():
...
+    mediation_fee = 130
+    action = ActionChannelSetFee(
+        canonical_identifier=state.canonical_identifier, mediation_fee=mediation_fee
+    )
+    result = handle_action_set_fee(state, action)
+    assert result.new_state.mediation_fee == mediation_fee
+    assert not result.events
+
+
+def make_hash_time_lock_state(amount):
```
Comment (palango) [inline raiden/tests/unit/transfer/test_channel.py]: nit: type annotations

File: raiden/blockchain_events_handler.py [hunk 6]
```diff
@@ -94,7 +96,14 @@ def handle_tokennetwork_new(raiden: "RaidenService", event: Event):  # pragma: n
...
 
-def handle_channel_new(raiden: "RaidenService", event: Event):
+def create_channel_new_state_change(
+    chain: "BlockChainService",
+    chain_id: typing.ChainID,
+    our_address: typing.Address,
+    payment_network_address: typing.PaymentNetworkAddress,
+    reveal_timeout: typing.BlockTimeout,
+    event: Event,
+) -> typing.Tuple[StateChange, typing.Optional[typing.Address], typing.Optional[FeeUpdate]]:
```
Comment (palango) [inline raiden/blockchain_events_handler.py]: This line is kind of ugly, maybe import the types directly.

File: raiden/tests/unit/test_blockchain_events_handler.py (hunk 1 above)
Comment (jomuel) [reply raiden/tests/unit/test_blockchain_events_handler.py]: I'd rather not call it `chain_state` as it's actually a dataclass containing a `chain_state` and other items. I'll go for `chain_state_setup` or something.

File: raiden/blockchain_events_handler.py [hunk 7]
```diff
@@ -150,7 +152,7 @@ def create_channel_new_state_change(
             reveal_timeout=reveal_timeout,
             payment_channel_proxy=channel_proxy,
             opened_block_number=block_number,
-            fee_schedule=replace(raiden.config["default_fee_schedule"]),
+            fee_schedule=fee_schedule,
```
Comment (palango) [inline raiden/blockchain_events_handler.py]: I think this is wrong, we want a copy of the default, not a reference to the default

Comment (jomuel) [reply raiden/blockchain_events_handler.py]: `fee_schedule` is a copy of the default, see the call to this function in `handle_channel_new` below.

**Pull request merged**